src/resolvers/Mutation/createAdvertisement.ts

```diff
@@ -1,14 +1,50 @@
 import type { MutationResolvers } from "../../types/generatedGraphQLTypes";
-import { Advertisement } from "../../models";
-// @ts-ignore
+import { Advertisement, User } from "../../models";
+import { errors, requestContext } from "../../libraries";
+import { USER_NOT_FOUND_ERROR } from "../../constants";
+import { uploadEncodedImage } from "../../utilities/encodedImageStorage/uploadEncodedImage";
+import { uploadEncodedVideo } from "../../utilities/encodedVideoStorage/uploadEncodedVideo";
+
 export const createAdvertisement: MutationResolvers["createAdvertisement"] =
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   async (_parent, args, _context) => {
+    // Get the current user
+    const currentUser = await User.findOne({
+      _id: _context.userId,
+    }).lean();
+
+    // Checks whether currentUser exists.
+    if (!currentUser) {
+      throw new errors.NotFoundError(
+        requestContext.translate(USER_NOT_FOUND_ERROR.MESSAGE),
+        USER_NOT_FOUND_ERROR.CODE,
+        USER_NOT_FOUND_ERROR.PARAM
+      );
+    }
+
+    let uploadMediaFile = null;
+
+    if (args.input.mediaFile) {
+      const dataUrlPrefix = "data:";
+      if (args.input.mediaFile.startsWith(dataUrlPrefix + "image/")) {
+        uploadMediaFile = await uploadEncodedImage(args.input.mediaFile, null);
+        console.log(uploadMediaFile, "Media File");
+      } else if (args.input.mediaFile.startsWith(dataUrlPrefix + "video/")) {
+        uploadMediaFile = await uploadEncodedVideo(args.input.mediaFile, null);
+      } else {
+        throw new Error("Unsupported file type.");
+      }
+    }
+
     // Creates new Ad.
-    const createdAd = await Advertisement.create({
-      ...args,
+    const createdAdvertisement = await Advertisement.create({
+      ...args.input,
+      mediaUrl: uploadMediaFile,
       creatorId: _context.userId,
+      organization: args.input.organizationId,
```

Comment:
`organizationId`

also you have to check for existence of an organization with `_id === args.input.organizationId`

mongodb doesn't have referential integrity, so it won't check if there's actually an organization with that id while creating the advertisement